velox/functions/prestosql/types/IPPrefixType.cpp

```diff
@@ -51,8 +80,114 @@ class IPPrefixCastOperator : public exec::CastOperator {
+
+      exec::StringWriter<false> resultWriter(flatResult, row);
+      if (v6Addr.isIPv4Mapped()) {
+        resultWriter.append(fmt::format(
+            "{}/{}", v6Addr.createIPv4().str(), prefix->valueAt(row)));
+      } else {
+        resultWriter.append(
+            fmt::format("{}/{}", v6Addr.str(), (uint8_t)prefix->valueAt(row)));
+      }
+      resultWriter.finalize();
+    });
+  }
+
+  static folly::small_vector<folly::StringPiece, 2> splitIpSlashCidr(
+      const folly::StringPiece& ipSlashCidr) {
+    folly::small_vector<folly::StringPiece, 2> vec;
+    folly::split('/', ipSlashCidr, vec);
+    return vec;
+  }
+
+  static void castFromString(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    int128_t intAddr;
+    folly::ByteArray16 addrBytes;
+    auto* rowResult = result.as<RowVector>();
+    const auto* ipAddressStrings = input.as<SimpleVector<StringView>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      auto ipAddressString = ipAddressStrings->valueAt(row);
+
+      // Folly allows for creation of networks without a "/" so check to make
+      // sure that we have one.
+      if (ipAddressString.str().find('/') == std::string::npos) {
+        context.setStatus(
+            row,
+            threadSkipErrorDetails() ? Status::UserError()
+                                     : Status::UserError(
+                                           "Cannot cast value to IPPREFIX: {}",
+                                           ipAddressString.str()));
+        return;
+      }
+
+      auto const maybeNet =
+          folly::IPAddress::tryCreateNetwork(ipAddressString, -1, false);
+
+      if (maybeNet.hasError()) {
+        context.setStatus(
+            row,
+            threadSkipErrorDetails() ? Status::UserError()
+                                     : Status::UserError(
+                                           "Cannot cast value to IPPREFIX: {}",
+                                           ipAddressString.str()));
+        return;
+      }
+
+      auto [ip, prefix] = maybeNet.value();
+      if (prefix > ((ip.isIPv4Mapped() || ip.isV4()) ? kIPV4Bits : kIPV6Bits)) {
+        context.setStatus(
+            row,
+            threadSkipErrorDetails() ? Status::UserError()
+                                     : Status::UserError(
+                                           "Cannot cast value to IPPREFIX: {}",
+                                           ipAddressString.str()));
+        return;
+      }
+
+      addrBytes = (ip.isIPv4Mapped() || ip.isV4())
+          ? folly::IPAddress::createIPv4(ip)
+                .mask(prefix)
+                .createIPv6()
+                .toByteArray()
+          : folly::IPAddress::createIPv6(ip).mask(prefix).toByteArray();
+
+      std::reverse(addrBytes.begin(), addrBytes.end());
+      memcpy(&intAddr, &addrBytes, ipaddress::kIPAddressBytes);
```

Comment:
Can we add a compile time check here that `sizeof(intAddr)` and `ipaddress::kIPAddressBytes` are the same?